### PR TITLE
State transition processing pipelines

### PR DESCRIPTION
**Pull request by rkapka**
**What type of PR is this?**

Other

**What does this PR do? Why is it needed?**

Another attempt at https://github.com/prysmaticlabs/prysm/pull/7191. I did not investigate what went wrong the first time, but implemented pipelines from scratch again.

**Which issues(s) does this PR fix?**

N/A

**Other notes for review**

I marked it as a WIP because we should be absolutely sure that this change doesn't break anything. I plan to discuss our testing strategy on Discord with @terencechain. Once testing is done, this PR can be reviewed and merged

### CODE REVIEW

File: beacon-chain/core/state/transition.go [hunk 1]
```diff
@@ -404,41 +420,46 @@ func ProcessBlockNoVerifyAnySig(
...
-	}
-
-	state, err = b.ProcessEth1DataInBlock(ctx, state, signed)
-	if err != nil {
-		traceutil.AnnotateError(span, err)
-		return nil, nil, errors.Wrap(err, "could not process eth1 data")
+	var bSet *bls.SignatureSet
+	var rSet *bls.SignatureSet
+	var aSet *bls.SignatureSet
+	var pipeline = []processFunc{
```
Comment (rkapka) [inline beacon-chain/core/state/transition.go:426]: Not the most elegant code, but I couldn't think of a better option to incorporate signature functions into the pipeline. To make the pipeline a package-level variable, it is necessary to move signature variables as well. I find this scoping clearer, but I don't have a strong opinion.

File: beacon-chain/core/state/transition.go [hunk 2]
```diff
@@ -404,41 +420,46 @@ func ProcessBlockNoVerifyAnySig(
...
+		b.ProcessEth1DataInBlock,
+		ProcessOperationsNoVerifyAttsSigs,
+		func(ctx context.Context,
+			state *stateTrie.BeaconState,
+			signed *ethpb.SignedBeaconBlock,
+		) (*stateTrie.BeaconState, error) {
+			var err error
+			aSet, err = b.AttestationSignatureSet(ctx, state, signed.Block.Body.Attestations)
+			return state, err
+		},
```
Comment (rkapka) [inline beacon-chain/core/state/transition.go:454]: Not the most elegant code, but I couldn't think of a better option to incorporate signature functions into the pipeline. To make the pipeline a package-level variable, it is necessary to move signature variables to package-level as well. I find this scoping clearer, but I don't have a strong opinion.

Comment (terencechain) [reply beacon-chain/core/state/transition.go:454]: I think this is worse than before. For that particular reason, I'm more inclined to leave `ProcessBlockNoVerifyAnySig` as it is. Having good readability out-weights the original intentions in my opinion...

**Comment by rauljordan** [comment]
Hey @rkapka whats the plan for this PR? Should we close and leave the branch open until we can further revisit it?

**Comment by rkapka** [comment]
Hey @rauljordan , this PR requires a long-running test. I am currently in the middle of syncing a node. This is very much active, even though it does not look this way. If everything works fine, this can get merged some time next week.

**Review by rauljordan — APPROVED** [review]
Cross-checked the operations are in order, could not see any problems. This is a fairly straightforward refactor. Approved

**Comment by rkapka** [comment]
Closing for the time being as we have to test this when proposing blocks full of deposits. Will reopen at a later date.

**Pull request closed without merging**